backend/schema/room.py

```diff
@@ -78,35 +78,69 @@ def validate_dm(cls, values):
         """
-        plugin_name = values.get("plugin_name")
-        room_members = values.get("room_members", {})
-        topic = values.get("topic")
-        description = values.get("description")
-        created_by = values.get("created_by")
-
-        if plugin_name == Plugin.DM and len(list(room_members.keys())) > 9:
-            raise HTTPException(
-                status_code=400, detail="Group DM cannot have more than 9 members"
-            )
-
-        if plugin_name == Plugin.DM and topic is not None:
-            raise HTTPException(status_code=400, detail="DM should not have a topic")
-
-        if plugin_name == Plugin.DM and description is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a description"
-            )
-
-        if plugin_name == Plugin.DM.value and created_by is not None:
-            raise HTTPException(
-                status_code=400, detail="DM should not have a created by"
-            )
+        if values["room_type"] != RoomType.CHANNEL:
+            room_type = values.get("room_type")
+            room_members = values.get("room_members", {})
+            topic = values.get("topic")
+            description = values.get("description")
+            org_id = values.get("org_id")
+
+            with concurrent.futures.ThreadPoolExecutor(1) as executor:
+                future = executor.submit(
+                    asyncio.run, get_org_rooms(org_id=org_id, room_type=room_type)
+                )
+                rooms = future.result()
+
+            if rooms is None:  # check if connection is avaliable
+                raise HTTPException(
+                    status_code=status.HTTP_424_FAILED_DEPENDENCY,
+                    detail="unable to read database",
+                )
+
+            if (
+                room_type == RoomType.GROUP_DM
+                and len(set(room_members.keys())) > 9
+                and len(set(room_members.keys())) < 2
+            ):
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Group DM cannot have more than 9 and less than 2 unique members",
+                )
+
+            if room_type == RoomType.DM and len(set(room_members.keys())) != 2:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="DM can only have 2 unique members",
+                )
+
+            if room_type in (RoomType.GROUP_DM, RoomType.DM):
+                for room in rooms:
+                    if set(room["room_members"].keys()) == set(room_members.keys()):
+                        raise HTTPException(
+                            status_code=status.HTTP_200_OK,
+                            detail={
+                                "message": "room already exists",
+                                "room_id": room["room_id"],
```

Comment:
Here, i think it should be:
```suggestion
                                "room_id": room["_id"],
```

Reply:
Thanks Fixed